### PR TITLE
sanity check metrics configuration

### PR DESCRIPTION
**Pull request by jbiseda**
#### Problem
Invalid metrics configuration can pollute metrics DB data. For example, a testnet validator could log to mainnet-beta database.

#### Summary of Changes
- Warn of a potential misconfiguration based on `ClusterType` and the database name from metrics configuration.
- Log the validator's `ClusterType` under `validator-new`

Fixes #

### CODE REVIEW

File: core/src/validator.rs
Comment (yhchiang-sol) [inline core/src/validator.rs:1257]: Not sure if `warn!` is strong enough.  Do we consider failing this unless a specific validator flag is set?

**Review by yhchiang-sol — COMMENTED** [review]
Thanks for improving this.  I can see the value of this PR.

File: core/src/validator.rs
Comment (jbiseda) [reply core/src/validator.rs:1257]: I considered this, but wasn't sure if it was too strong of a requirement. Since you also mentioned this I'll go this route.

File: metrics/src/metrics.rs
Comment (t-nelson) [inline metrics/src/metrics.rs:413]: it should be ok if metrics aren't configured at all, right?  might be time for a proper error type here (separate pr?) rather than the lazy `Result<T, String>`. should be pretty easy to swap out with an enum and `thiserror` crate

File: core/src/validator.rs
Comment (t-nelson) [reply core/src/validator.rs:1257]: i'd go so hard as to `std::process:exit(1)`. these people have fooled me too many times :wink:

**Review by t-nelson — COMMENTED** [review]
great initiative!

**Review by yhchiang-sol — APPROVED** [review]
Thanks for working on this!
LGTM.  Please also have @t-nelson's approval before merging the PR.

File: upload-perf/src/upload-perf.rs
Comment (t-nelson) [inline upload-perf/src/upload-perf.rs:17]: `thiserror` provides `Display`, can `?` not autoconvert?  maybe needs explicit `impl Into<String> for MetricsError`... 🤔

File: validator/src/cli.rs
Comment (t-nelson) [inline validator/src/cli.rs:560]: what's the usecase here? seems like the logic will already gracefully handle joining a public cluster with a private db

**Review by t-nelson — COMMENTED** [review]
much nicer with the custom error!

File: validator/src/cli.rs
Comment (jbiseda) [reply validator/src/cli.rs:560]: Logging to a db on a private host named `tds` for mainnet? 🤡

@t-nelson rip it out?

File: upload-perf/src/upload-perf.rs
Comment (jbiseda) [reply upload-perf/src/upload-perf.rs]: added ` From<MetricsError> for String`

File: validator/src/cli.rs
Comment (t-nelson) [reply validator/src/cli.rs:560]: yeah let's start w/o it and see if there's demand.  these sanity check skipping flags have hurt us more than they've helped in the past

Comment (jbiseda) [reply validator/src/cli.rs]: 🪓

**Review by t-nelson — APPROVED** [review]
:shipit:

**Pull request merged**
